pymatgen/io/qchem/inputs.py

```diff
@@ -458,7 +458,7 @@ def find_sections(string: str) -> List:
         return sections
 
     @staticmethod
-    def read_molecule(string: str) -> Molecule:
+    def read_molecule(string: str) -> Union[Molecule, str]:
```

Comment:
This cannot be the correct type. You actually return a Molecule, not a string.

Reply:
It depends. It's possible for the molecule section to just be "read", meaning you're reading in the molecule from a previous calculation.

Reply:
Then it should be Optional[Molecule]. Not str. Reading string to return a string is bad.

Reply:
I suppose we could do Optional, but "read" is really the correct value. It doesn't just indicate the lack of a molecule, but that the molecule is being read in from somewhere else.

Reply:
Following that through, the type hint for `QCMolecule` `__init__` would be incorrect too since that says it only accepts `Molecule` not `"read"`.

Btw, if there's only one possible string, the correct hint is `Literal["read"]`.

Reply:
I do also concur that `None` is a more natural value than `"read"`, but I'm not familiar with QChem, so perhaps `"read"` makes more sense to QChem users?

Reply:
Yup, that makes sense, given its usage. Just make sure the `QCMolecule.__init__` has a consistent type hint, since currently it states it can only accept `Molecule`, when it seems like it could be passed `"read"` instead.

Reply:
Ooh, unfortunate. Looks like Literal was only added in Python 3.8 ([see here](https://www.python.org/dev/peps/pep-0586/)).

Since that's not an option, are we okay with "str" (there is a check on that in QCInput.__init__ to make sure that, if it's a string, it's "read")? Or should I just go with Optional at that point?

Also, not sure what you're referring to by QCMolecule, @mkhorton. I don't think that's a class in pymatgen?

Reply:
Meant `QCInput`, apologies.

There's a backport with typing-extensions, we can add it -- I think it's important enough to do so.

Reply:
Actually, @shyuep already added typing-extensions two days ago, so we're good. I just added to the setup.py too and made it only install for older versions of Python.

Reply:
I'll add that to requirements.txt and make the change.